### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by kpetridis24** [networkx/networkx#5569] [Ideas]
Any opinions on following a more organized and structured approach in testing using [Unit Tests](https://docs.python.org/3/library/unittest.html) instead of regular python assertions?

**Comment by MridulS** [comment]
We already use [pytest](https://docs.pytest.org) for creating and setting up our testing, which is much more powerful (IMO) than the inbuilt unit tests from Python.

**Reply by rossbar** [reply]
Agreed - the pytest framework is much more general and has many more features. In fact, it supports `unittest`-style tests, but I'd prefer matching the patterns already used throughout the test suite.
